### PR TITLE
f/555 historic data range

### LINKED ISSUE #555: change requested data range
**Is your feature request related to a problem? Please describe:**  
we don't really use the historical data. We don't show other time intervals then 30seconds

**Please describe the solution you'd like:** 
The user should have the possibility to change the requested time range

**Affected Component(s):**  
Frontend

**Additional context:**

### PR DESCRIPTION
**Pull request by MaxSchneider1337**
# Resolves #555

**Does your pull request add a feature? Please describe:**  
change requested historic data range 

**Affected Component(s):**  
Frontend

**Additional context:**  
Add any other context or screenshots about the pull request here.

### CODE REVIEW

File: hyrisecockpit/frontend/src/services/metricService.ts [hunk 1]
```diff
@@ -146,17 +146,19 @@ export function useMetricService(metrics: Metric[]): MetricService {
   function handleCurrentDataPoints<T>(data: T[], newData: T[]): T[] {
     const dataCopy = data;
     newData.forEach((entry: T) => {
-      if (dataCopy.length > 29) {
-        dataCopy.shift();
-      }
+      // if (dataCopy.length > 29) {
```
Comment (MaxSchneider1337) [inline hyrisecockpit/frontend/src/services/metricService.ts]: @schTi any ideas how to access the metricController data her to get the historic data range?

**Comment by MaxSchneider1337** [comment]
overview timestamps style bugs when range is too high

File: hyrisecockpit/frontend/src/services/metricService.ts (hunk 1 above)
Comment (schTi) [reply hyrisecockpit/frontend/src/services/metricService.ts]: Just pass it as additional parameter inside the start function?

File: hyrisecockpit/frontend/src/components/charts/Linechart.vue [hunk 2]
```diff
@@ -52,7 +53,7 @@ export default defineComponent({
       Plotly.newPlot(
         props.graphId,
         getDatasets(),
-        getLayout(props.maxValue),
+        getLayout(props.maxValue, getHistoricRangeMinutes() * 60),
```
Comment (schTi) [inline hyrisecockpit/frontend/src/components/charts/Linechart.vue]: You always need to multiply with 60, so why not wrap this in a function?

**Review by schTi — CHANGES_REQUESTED** [review]
Good stuff, just one thing

File: hyrisecockpit/frontend/src/components/charts/Linechart.vue (hunk 2 above)
Comment (MaxSchneider1337) [reply hyrisecockpit/frontend/src/components/charts/Linechart.vue]: good point. this even stood on my list, but I forgot it.

File: hyrisecockpit/frontend/src/controller/metricController.ts [hunk 3]
```diff
@@ -15,10 +15,10 @@ type Interval = {
 export function useMetricController(): MetricController {
   const { subSeconds } = useFormatting();
 
-  let historicRangeMinutes = 0.5;
+  let historicRangeSeconds = 30;
 
-  function getHistoricRangeMinutes(): number {
-    return historicRangeMinutes;
+  function gethistoricRangeSeconds(): number {
```
Comment (schTi) [inline hyrisecockpit/frontend/src/controller/metricController.ts]: should be getHistoricRangeSeconds

**Review by schTi — CHANGES_REQUESTED** [review]

**Review by schTi — APPROVED** [review]
Good job

**Pull request merged**